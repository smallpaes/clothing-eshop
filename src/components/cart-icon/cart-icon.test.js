import { shallow } from 'enzyme';
import CartIcon from './cart-icon.component';

describe('CartIcon Component', () => {
  let wrapper;
  let mockToggleCartHidden;
  beforeEach(() => {
    mockToggleCartHidden = jest.fn();
    const mockProps = {
      itemCount: 0,
      toggleCartHidden: mockToggleCartHidden
    };
    wrapper = shallow(<CartIcon {...mockProps} />);
  });


  it('Should render CartIcon component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render itemCount correctly', () => {
    const renderedText = wrapper.find('ItemCountContainer').text();
    expect(renderedText).toBe('0');
  });

  it('Should call toggleCartHidden when icon is clicked', () => {
    const cartIconNode = wrapper.find('CartIconContainer');
    cartIconNode.simulate('click');
    expect(mockToggleCartHidden.mock.calls.length).toBe(1);
  });
});