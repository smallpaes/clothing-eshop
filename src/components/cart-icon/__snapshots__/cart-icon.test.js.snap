// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon Component Should render CartIcon component 1`] = `ShallowWrapper {}`;
